nbs/explains/background_tasks.ipynb

```diff
@@ -0,0 +1,296 @@
+    "\n",
+    "app, rt = fast_app()\n",
+    "\n",
+    "def counter(loops:int): # <1>\n",
+    "    \"\"\"Slowly print integers to the terminal\"\"\"\n",
+    "    for i in range(loops):\n",
+    "        print(i)\n",
+    "        sleep(i)\n",
+    "\n",
+    "@rt\n",
+    "def index():\n",
+    "    task = BackgroundTask(counter, loops=5)  # <2>\n",
+    "    return Titled('Simple Background Task Example'), task # <3>\n",
+    "\n",
+    "serve()\n",
+    "```\n",
+    "\n",
+    "1. `counter` is our task function. There is nothing special about it, although it is a good practice for its arguments to be serializable as JSON\n",
+    "2. We use `starlette.background.BackgroundTask` to turn `counter()` into a background task\n",
+    "3. To add a background task to a handler, we add it to the return values at the top level of the response."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A more realistic example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's imagine that we are accessing a slow-to-process critical service. We don't want our users to have to wait. While we could set up SSE to notify on completion, instead we decide to periodically check to see if the status of their record has changed."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Simulated Slow API Service"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, create a very simple slow timestamp API. All it does is stall requests for a few seconds before returning JSON containing timestamps."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# slow_api.py\n",
+    "from fasthtml.common import *\n",
+    "from time import sleep, time\n",
+    "\n",
+    "app, rt = fast_app()\n",
+    "\n",
+    "@rt('/slow')\n",
+    "def slow(ts: int):\n",
+    "    sleep(3) # <1>\n",
+    "    return dict(request_time=ts, response_time=int(time())) # <2>\n",
+    "\n",
+    "serve(port=8123)\n",
+    "```\n",
+    "\n",
+    "1. This represents slow processing.\n",
+    "2. Returns both the task's original timestamp and the time after completion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Main FastHTML app"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's create a user-facing app that uses this API to fetch the timestamp from the glacially slow service."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# main.py\n",
+    "from fasthtml.common import *\n",
+    "from fastlite import *\n",
+    "from starlette.background import BackgroundTask\n",
+    "import time\n",
+    "import httpx\n",
+    "\n",
+    "app, rt = fast_app()\n",
+    "\n",
+    "db = database(':memory:')\n",
+    "\n",
```

Comment:
Place class on 1 line. No need for 3 separate callouts here IMO - it's clear enough and isn't specific to what you're teaching.